drools-eclipse/org.drools.eclipse/src/main/java/org/drools/eclipse/debug/AuditView.java

```diff
@@ -146,17 +148,29 @@ public void refresh() {
         } catch (Throwable t) {
             DroolsEclipsePlugin.log(t);
         }
-        if (phreak) {
-            getViewer().setInput(createPhreakEventList(eventList));
+        if (reteoo) {
+            getViewer().setInput(createEventList(eventList));
         } else if (drools4) {
             getViewer().setInput(createDrools4EventList(eventList));
         } else {
-            getViewer().setInput(createEventList(eventList));
+            getViewer().setInput(createPhreakEventList(eventList));
         }
         // TODO: this is necessary because otherwise, the show cause action
         // cannot find the cause event if it hasn't been shown yet
         ((TreeViewer) getViewer()).expandAll();
     }
+
+    private boolean isReteoo(WorkingMemoryLog log) {
+        // From Drools 7 phreak is the only available engine.
+        // Tries to check via reflection if the WorkingMemoryLog class has a getEngine() method
+        // (only available on drools 6) and use it if so.
+        try {
+            Method m = log.getClass().getMethod( "getEngine" );
+            return ((String)m.invoke( log )).equalsIgnoreCase("RETEOO");
+        } catch (Exception e) {
```

Comment:
Would it make sense to catch more specific exception like `NoSuchMethodException`?